crates/snp_validation/src/lib.rs

```diff
@@ -287,96 +309,143 @@ impl Snp {
         // TODO: additional field validations.
 
         // Should only be version 2
-        if report.body.version != 2 {
-            return Err(anyhow!("snp report is an unknown version"));
-        }
+        ensure!(
+            report.body.version == 2,
+            "snp report has an unknown version"
+        );
 
         // Check policy
-        if !report.body.policy.flags.contains(PolicyFlags::Reserved) {
-            return Err(anyhow!("snp guest policy mandatory reserved flag not set"));
-        }
-
-        if report.body.policy.flags.contains(PolicyFlags::MigrateMA) {
-            return Err(anyhow!("snp guest policy migration flag was set"));
+        ensure!(
+            report.body.policy.flags.contains(PolicyFlags::Reserved),
+            "snp guest policy mandatory reserved flag not set"
+        );
+
+        // Enarx doesn't support migration to another machine
+        ensure!(
+            !report.body.policy.flags.contains(PolicyFlags::MigrateMA),
+            "snp guest policy migration flag was set"
+        );
+
+        // Only check major.minor, as there isn't a revision or patch version in the attestation.
+        if report.body.policy.abi_major > report.body.current_major {
+            bail!("snp policy has higher abi major than firmware");
+        } else if report.body.policy.abi_minor > report.body.current_minor {
+            bail!("snp policy has higher abi minor than firmware");
         }
 
         // Check reserved fields
-        if report.body.rsvd1 != 0 || report.body.rsvd3 != 0 || report.body.rsvd4 != 0 {
-            return Err(anyhow!("snp report reserved fields were set"));
-        }
+        ensure!(
+            report.body.rsvd1 == 0,
+            "snp report reserved field 1 was set"
+        );
+
+        ensure!(
+            report.body.rsvd2 == [0u8; 24],
+            "snp report reserved field 2 was set"
+        );
+
+        ensure!(
+            report.body.rsvd4 == 0,
+            "snp report reserved field 3 was set"
+        );
+
+        ensure!(
+            report.body.rsvd4 == 0,
+            "snp report reserved field 4 was set"
+        );
+
+        ensure!(
+            report.body.rsvd5 == [0u8; 168],
+            "snp report reserved field 5 was set"
+        );
+
+        ensure!(
+            report.body.policy.rsvd == [0u8; 5],
+            "snp report policy reserved fields were set"
+        );
+
+        ensure!(
+            report.body.plat_info.rsvd == [0u8; 7],
+            "snp report platform_info reserved fields were set"
+        );
 
-        for value in report.body.rsvd2 {
-            if value != 0 {
-                return Err(anyhow!("snp report reserved fields were set"));
-            }
-        }
+        // Check fields not set by Enarx
+        ensure!(
+            report.body.host_data == [0u8; 32],
```

Comment:
Isn't `u8` suffix here(and other occurences)  redundant?

Reply:
Removed